packages/block-library/src/post-content/edit.js

```diff
@@ -13,28 +14,45 @@ import {
 	Warning,
 } from '@wordpress/block-editor';
 import { useEntityProp, useEntityBlockEditor } from '@wordpress/core-data';
+import { useMemo } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import { useCanEditEntity } from '../utils/hooks';
 
-function ReadOnlyContent( { userCanEdit, postType, postId } ) {
+function ReadOnlyContent( { layout, userCanEdit, postType, postId } ) {
 	const [ , , content ] = useEntityProp(
 		'postType',
 		postType,
 		'content',
 		postId
 	);
 	const blockProps = useBlockProps();
+
+	const themeSupportsLayout = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
+		return getSettings()?.supportsLayout;
+	}, [] );
+	const defaultLayout = useSetting( 'layout' ) || {};
+	const usedLayout = !! layout && layout.inherit ? defaultLayout : layout;
+
+	const rawContent = content?.raw;
+	const blocks = useMemo( () => {
+		return rawContent ? parse( rawContent ) : [];
+	}, [ rawContent ] );
+
+	const blockPreviewProps = useBlockPreview( blockProps, {
+		blocks,
+		__experimentalLayout: themeSupportsLayout ? usedLayout : undefined,
```

Comment:
Is the logic for the layout something that could be moved to `useBlockPreview`? You could potentially read the `layout` attribute from the block edit context.

Reply:
In theory, `useBlockProps` can be called inside `useBlockPreview`, too. We might need the same hook for Comment Content block one day so I'm thinking about how to make this hook easier to use in other places.

Reply:
That's a neat idea — I think for the moment, I'm leaning toward keeping this particular layout logic outside of the `useBlockPreview` hook, in case it's possible for us to use the hook in other circumstances where `useBlockProps` isn't appropriate. I'm thinking of arbitrary block previews that are used elsewhere in the repo, for example in the block switcher or for block patterns. They currently use the `BlockPreview` component directly, but it could be good to retain some flexibility for the hook in case we wanted to roll it out there, (or internally in the `BlockPreview` component itself?

Reply:
You are absolutely right. I didn't think about the original use case of `<BlockPreview />` that was hidden in the changes for this PR. I guess the usage of the preview inside the `edit` is rather an edge case so we shouldn't expect that the block edit context is set here :+1: